cmd/river/riverbench/river_bench.go

```diff
@@ -132,13 +145,30 @@ func (b *Benchmarker[TTx]) Run(ctx context.Context) error {
 			case <-shutdown:
 				return
 
-			case <-subscribeChan:
-				numJobsLeft.Add(-1)
-				numJobsWorked := numJobsWorked.Add(1)
+			case event := <-subscribeChan:
+				if event == nil { // Closed channel.
+					b.logger.InfoContext(ctx, "Subscription channel closed")
+					return
+				}
+
+				switch {
+				case event.Kind == river.EventKindJobCancelled:
+					fallthrough
+
+				case event.Kind == river.EventKindJobCompleted:
+					fallthrough
+
+				// Only count a job as complete if it failed for the last time.
+				// We don't expect benchmark jobs to ever fail, so this extra
+				// attention to detail is here, but shouldn't be needed.
```

Comment:
should we count and log these for debug purposes?

Reply:
Sure. Added this as an error log since it should never, ever happen. The errors will be visible even without passing the verbose flag.